Raven.Database/Commercial/ValidateLicense.cs

```diff
@@ -77,105 +79,112 @@ public void ForceExecute(InMemoryRavenConfiguration config)
 
         private void ExecuteInternal(InMemoryRavenConfiguration config, bool firstTime = false, bool forceUpdate = false)
         {
-            var licensePath = GetLicensePath(config);
-            var licenseText = GetLicenseText(config);
-
-            if (TryLoadLicense(config) == false)
-                return;
-
-            string errorMessage = string.Empty;
-            try
+            // Thread will execute code only if validationLockAcquired was 0 upon entering method, else it will return
+            if (Interlocked.CompareExchange(ref validationLockAcquired, 1, 0) == 0)
```

Comment:
No, this will fail if there was an exception in this method, which will cause it to skip setting `validationLockAcquired` to zero, resulting in no one ever executing this code again.